Makefile

```diff
@@ -77,166 +50,52 @@ endif
 C_O_FILES := $(foreach f,$(C_FILES:.c=.o),$(BUILD_DIR)/$f)
 S_O_FILES := $(foreach f,$(S_FILES:.s=.o),$(BUILD_DIR)/$f)
 O_FILES   := $(S_O_FILES) $(C_O_FILES)
-# Because we patch the object file timestamps, we can't use them as the targets since they'll always be older than the C file
-# Therefore instead we use marker files that have actual timestamps as the dependencies for the archive
-C_MARKER_FILES := $(C_O_FILES:.o=.marker)
-S_MARKER_FILES := $(S_O_FILES:.o=.marker)
-S_MARKER_FILES := $(filter-out $(MDEBUG_FILES),$(S_MARKER_FILES))
-MARKER_FILES   := $(C_MARKER_FILES) $(S_MARKER_FILES) $(MDEBUG_FILES)
-
-ifneq ($(COMPARE),0)
-COMPARE_OBJ = cmp $(BASE_DIR)/$(@F:.marker=.o) $(@:.marker=.o) && echo "$(@:.marker=.o): OK"
-COMPARE_AR = cmp $(BASE_AR) $@ && echo "$@: OK"
-ifeq ($(COMPILER),ido)
-COMPARE_OBJ = $(CROSS)objcopy -p --strip-debug $(WORKING_DIR)/$(@:.marker=.o) $(WORKING_DIR)/$(@:.marker=.cmp.o) && \
-              cmp $(BASE_DIR)/.cmp/$(@F:.marker=.cmp.o) $(WORKING_DIR)/$(@:.marker=.cmp.o) && echo "$(@:.marker=.o): OK"
-COMPARE_AR = echo "$@: Cannot compare archive currently"
-endif
-else
-COMPARE_OBJ :=
-COMPARE_AR :=
-AR_OLD := $(AR)
-endif
-
-BASE_OBJS := $(wildcard $(BASE_DIR)/*.o)
-
-# Check to make sure the current version has been set up
-ifneq ($(COMPARE),0)
-ifeq ($(BASE_OBJS),)
-# Ignore this check if the user is currently running setup, clean or distclean
-ifeq ($(filter $(MAKECMDGOALS),setup clean distclean),)
-$(error Current version ($(TARGET) 2.0$(VERSION)) has not been setup!)
-endif
-endif
-endif
 
 AR_OBJECTS := $(shell cat base/$(VERSION)/$(TARGET).txt)
 # If the version and target doesn't have a text file yet, resort back to using the base archive to get objects
-ifeq ($(AR_OBJECTS),)
-AR_OBJECTS := $(shell ar t $(BASE_AR))
-endif
-
 
 # Try to find a file corresponding to an archive file in src/ or the base directory, prioritizing src then the original file
-AR_ORDER = $(foreach f,$(AR_OBJECTS),$(shell find $(BUILD_DIR)/src $(BASE_DIR) -iname $f -type f -print -quit))
-MATCHED_OBJS = $(filter-out $(BASE_DIR)/%,$(AR_ORDER))
-UNMATCHED_OBJS = $(filter-out $(MATCHED_OBJS),$(AR_ORDER))
-NUM_OBJS = $(words $(AR_ORDER))
-NUM_OBJS_MATCHED = $(words $(MATCHED_OBJS))
-NUM_OBJS_UNMATCHED = $(words $(UNMATCHED_OBJS))
+AR_ORDER = $(foreach f,$(AR_OBJECTS),$(shell find $(BUILD_DIR)/src -iname $f -type f -print -quit))
```

Comment:
The ar order is a matching artifact to match the original order of files in static archives found on CDs. It shouldn't be needed for a modern build.